statsmodels/multivariate/manova.py

```diff
@@ -0,0 +1,301 @@
+                     tests=None, *args, **kwargs):
+        mod = super(MANOVA, cls).from_formula(formula, data,
+                                              subset=subset,
+                                              drop_cols=drop_cols,
+                                              tests=tests,
+                                              *args, **kwargs)
+        terms = mod.design_info.term_name_slices
+        hypothesis = []
+        for key in terms:
+            L_contrast = np.eye(mod.exog.shape[1])[terms[key], :]
+            hypothesis.append((key, L_contrast, None))
+        return mod, hypothesis
+
+    def test(self, H=None):
+        """
+        Testing the genernal hypothesis
+            L * B * M = 0
+        for each tuple (name, L, M) in `H` where B is the coefficient matrix
+        for the linear model Y = X * B
+
+        Parameters
+        ----------
+        H: A list of tuples
+           Hypothesis to be tested. Each element is a tuple (name, L, M)
+           containing a string `name`, the contrast matrix L and the transform
+           matrix M for transforming dependent variables, respectively. If M is
+           `None`, it would be set to an identity matrix (i.e. no dependent
+           variable transformation).
+           If `H` is None, the effect of each independent variable on the
+           dependent variables will be tested.
+
+        Returns
+        -------
+        results: a list of tuples (name, manova_table) for each tuple in `H`
+            MANOVA results
+
+        """
+        if H is None:
+            H = []
+            for i in range(self.exog.shape[1]):
+                name = 'x%d' % (i)
+                L = np.zeros([1, self.exog.shape[1]])
+                L[i] = 1
+                H.append([name, L, None])
+
+        results = []
+        for name, L, M in H:
+            if len(L.shape) != 2:
+                raise ValueError('Contrast matrix L must be a 2-d array!')
+            if L.shape[1] != self.exog.shape[1]:
+                raise ValueError('Contrast matrix L should have the same '
+                                 'number of columns as exog! %d != %d' %
+                                 (L.shape[1], self.exog.shape[1]))
+            if M is not None:
+                if len(M.shape) != 2:
+                    raise ValueError('Transform matrix M must be a 2-d array!')
+                if M.shape[0] != self.endog.shape[1]:
+                    raise ValueError('Transform matrix M should have the same '
+                                     'number of rows as the number of columns '
+                                     'of endog! %d != %d' %
+                                     (M.shape[0], self.exog.shape[1]))
+            fit_output = (self.reg_coeffs, self.df_resid, self.inv_cov_,
+                          self.YYBXXB_)
+            manova_table = test_manova(fit_output, L, M)
+            results.append((name, manova_table))
+        return results
```

Comment:
It's better if this returns some kind of results instance, e.g. t_test and similar return a ContrastResults.
The list access to the results is a bit cumbersome for users, and we cannot easily change the returns by adding e.g. new features.
For example a MANOVAResults instance could have a nice `__str__` or `__repr__` to print a summary of the results.